### ISSUE TITLE
Implement pipe in terms of flow

### ISSUE THREAD

**Issue by OliverJAsh** [gcanti/fp-ts#1229]
### Current Behavior

When I was looking through the output of webpack I noticed that we're basically writing the same function twice. If we implement `pipe` in terms of `flow` then we can save a few bytes. (Alternatively you could use `pipe-ts` which does this.)

(I had a go at doing this myself but got stuck with the types.)

### Your environment

| Software   | Version(s) |
| ---------- | ---------- |
| fp-ts      |            |
| TypeScript |            |

**Comment by gcanti** [comment]
@OliverJAsh https://github.com/gcanti/fp-ts/blob/master/perf/pipe.ts

```
pipe x 13,635,124 ops/sec ±0.61% (90 runs sampled)
pipe implemented in terms of flow x 3,163,124 ops/sec ±0.35% (87 runs sampled)
```

**Comment by OliverJAsh** [comment]
Interesting. I guess it's a trade off between runtime perf and bundle size. Thanks for sharing!
